libraries/azure-app-configuration-importer/src/internal/parsers/defaultConfigurationSettingsConverter.ts

```diff
@@ -272,4 +402,49 @@ class FeatureFlagConfigurationSettingsConverter implements ConfigurationSettings
 
     return lowerCaseFilters;
   }
+
+  private validateRequirementType(requirement_type: any, featureFlagName: string) {
+    if(!Object.values(RequirementType).includes(requirement_type)){
+      throw new ArgumentError(
+        `This feature flag '${featureFlagName}' must have any/all requirement type.`
+      );
+    }
+
+    return true;
+  }
+
+  private getMsFmSchemaFeatureFlags(config: object): any {
+    const msFeatureManagementKeyWord = Constants.FeatureManagementKeyWords[3];
+    const featureManagementKey =  msFeatureManagementKeyWord as keyof object;
+    const featureManagementSection = config[featureManagementKey];
+
+    if (typeof featureManagementSection !== "object") {
+      throw new ArgumentError(`The ${msFeatureManagementKeyWord} section must be an object.`);
+    }
+
+    if (!Array.isArray(featureManagementSection[Constants.FeatureFlagsKeyWord])) {
+      throw new ArgumentError(`The ${Constants.FeatureFlagsKeyWord} key within ${msFeatureManagementKeyWord} must be an array.`);
+    }
+
+    return featureManagementSection[Constants.FeatureFlagsKeyWord];
+  }
+
+  private getLegacySchemaFeatureFlags(config: object): any {
+    const msFeatureManagementKeyWord = Constants.FeatureManagementKeyWords[3];
+    const featureManagementSection = config[this.legacySchemaFeatureManagementKeyWord as keyof object];
+
+    if (typeof featureManagementSection !== "object") {
+      throw new ArgumentError(`The ${this.legacySchemaFeatureManagementKeyWord} section must be an object.`);
+    }
+
+    if (this.legacySchemaFeatureManagementKeyWord === msFeatureManagementKeyWord) { //legacy schema might be nested within msFmSchema
+      const { feature_flags, ...rest } = featureManagementSection as { [key: string]: any };
+      if (Object.keys(rest).length > 0) {
```

Comment:
For the if block on line 440 if legacy schema is nested meaning we have something like this 
```{
  "feature_management": {
    "feature_flags": [
      {
        "id": "FeatureZ",
        "enabled": true
      },
      {
        "id": "FeatureY",
        "enabled": false
      }
    ],
      "FeatureX": true,
      "FeatureY": true
    }
}
```
since we return `rest` ? does it mean we ignore `feature_flags`

Also does `rest` here refer to the the traditional feature flags that follow legacy schema ?

Reply:
Yes we will ignore the part with `feature_flags`. Yes rest refers to the traditional flags I could rename it better though?
Maybe `const { feature_flags, ...legacyflags } ` what do you think?

Reply:
renamed to dotnetSchemaFlags